README.md

```diff
@@ -2,19 +2,22 @@
 
 The prom-label-proxy enforces a given label in a given PromQL proxy.
 
-This proxy does not perform authentication or authorization, this has to happen
-before the request reaches this proxy. The
-[kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) is an example for
-such an additional building block.
-
+This proxy does not perform authentication or authorization, this has to happen before the request reaches this proxy. The [kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) is an example for such an additional building block.
 
 Risks outside the scope of this project:
 
-- If a tenant controls its scrape target configuration the tenant can set
-  arbitrary labels via its relabelling configuration, thereby being able to
-  pollute other tenant's metrics.
+- If a tenant controls its scrape target configuration the tenant can set arbitrary labels via its relabelling configuration, thereby being able to pollute other tenant's metrics.
+
+- If the ingestion configuration [honor_labels](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config) is set for a tenant's target, that target can pollute other tenant's metrics as Prometheus respects any labels exposed by the target.
+
+## How does this project work?
+
+What this proxy does is it proxies the `/federate`, `/api/v1/query`, `/api/v1/query_range` prometheus endpoints and ensures that a label particular label is enforced in the particular query.
+
+In the case of the federate endpoint, it ensures that all selectors passed as matchers to the federate endpoint _must_ contain that exact match of the particular label (and throws away all other matchers for the label). For the two query endpoints, it parses the PromQL expression and modifies all selectors in the same way. The label-key is configured as a flag on the binary and label-value is passed as a query parameter.
+
```

Comment:
Would be nice to add an more hands-on example, something like:

> Here's an example: We are running our application in the namespace `app` and our application developers should only see metrics for their namespace. This proxy can be configured to enforce that all queries sent to Prometheus always have `{namespace="app"}` as part of their label selectors. Once this label selector is enforced, the returned metrics are only for the developer's `app` namespace.

Reply:
Done